src/js/components/form-elements/DateFilter/DateFilter.jsx

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from 'react';
+
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import DatePicker from 'react-datepicker';
+import { RiCalendarLine, RiCloseLine } from 'react-icons/ri';
+
+import 'react-datepicker/dist/react-datepicker.css';
+import 'components/form-elements/DateFilter/DateFilter.scss';
+
+const CustomInput = React.forwardRef((props, ref) => {
+  const {
+    onClick, title, value, placeholder, onClear,
+  } = props;
+  return (
+    <button
+      ref={ref}
+      className="d-flex flex-row date-picker__input-wrapper"
+      onClick={onClick}
+    >
+      <span className="flex-grow-1 date-picker__input">
+        <span>{title}</span>
+        <span>{value || placeholder}</span>
+      </span>
+      <div className="date-picker__icon-wrapper">
+        {
+          value
+            ? <button className="date-picker__icon" onClick={onClear}><RiCloseLine /></button>
+            : <div className="date-picker__icon"><RiCalendarLine /></div>
+        }
+      </div>
+    </button>
+  );
+});
+
+
+const DateFilter = (props) => {
+  const {
+    value, onChange, dateFormat, placeholder, label, timeFormat,
+  } = props;
+  const [isFocused, setIsFocused] = useState(false);
+
+  const onChangeHandler = (date) => {
+    const val = !date || typeof date === 'string' ? date : date.format(dateFormat);
```

Comment:
I assume it is working how it should be, but could you explain what it is exactly doing?

Reply:
I copied this from the **DateField** component  and looking at it again it looks unnecessary complicated so in the new commit I simplified it.